network-protocols.md

```diff
@@ -276,9 +293,37 @@ If `response.State` is `Accepted` then the client should proceed to transfer the
 Next, when the miner receives all the data and validates it, they set the deals state to `Staged`. When the sector gets sealed, and the commitment is posted on chain, the state gets set to `Complete` and the deals `PieceConfirmation` field should be set to the appropriate values.
 
 ```go
-func OnDataReceived() {
-    // TODO: document process for updating deal state and
-    // starting seal after data is received
+func OnDataReceived(prop StorageDealProposal) {
+  if !ValidatePieceTranslation(prop.PieceRef, prop.SerializationMode, prop.TranslatedRef) {
+    resp := StorageDealResponse{
+        State: Rejected,
+        Proposal: prop.Cid(),
+      	Message: "TranslatedRef was invalid, reconstructed data did not match",
+    }
+
+    miner.Sign(resp)
+    miner.SetDealState(resp)
+    return
+  }
+  
+  // TODO: is TranslatedRef actually needed? How does it tie in?
+  pieceConfirmation := SectorBuilder.PackData(prop.PieceRef, prop.SerializationMode) 
```

Comment:
At this point, the `SectorBuilder` will have added a user's piece to a yet-to-be-sealed sector. The miner doesn't have to seal it immediately. How will the miner be able to generate a piece inclusion proof for this piece if the miner hasn't yet filled up and kicked off the sealing process?